tests/test_generator.py

```diff
@@ -32,6 +40,38 @@ def test_can_generate_azure_pipeline(dummy_pipeline, dummy_plugin_config):
         ), "Invalid docker image set on commands"
 
 
+def test_azure_pipeline_with_different_compute(
+    dummy_pipeline_compute_tag, dummy_plugin_config
+):
+    """
+    Test that when a Node in an Azure Pipeline is tagged with a compute tag
+    this gets passed through to the generated azure pipeline
+    """
+    with patch.object(
+        AzureMLPipelineGenerator,
+        "get_kedro_pipeline",
+        return_value=dummy_pipeline_compute_tag,
+    ):
+        env_name = "unit_test_env"
+        docker_image = "unit_test/docker_image:latest"
+        generator = AzureMLPipelineGenerator(
+            "dummy_pipeline_compute_tag",
+            env_name,
+            dummy_plugin_config,
+            docker_image=docker_image,
+        )
+
+        az_pipeline = generator.generate()
+        for node in dummy_pipeline_compute_tag.nodes:
+            if node.tags:
+                for tag in node.tags:
+                    if "azureml.compute" in tag:
```

Comment:
Please remember to update the tests once you finish the changes to the kedro tag -> azure compute mapping.